### ISSUE TITLE
Valid link in e-mail was disallowed

### ISSUE THREAD

**Issue by skaizun** [AdguardTeam/AdguardBrowserExtension#1939]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [X ] I am running the latest version
- [X ] I checked the documentation and found no answer
- [X ] I checked to make sure that this issue has not already been filed

### Issue Details

- **AdGuard version:**
 - 3.6.17 (latest)

- **Browser and version:**
- Chrome Version 96.0.4664.110 (latest)

- **Operating system and version:**
  - Windows 10 (latest)
  
- **Filters you use in AdGuard:**
  - default
  
- **Stealth Mode settings:**
  - not enabled
  
- **Other browser extensions:**
  - too many to list, but, I disabled Adguard first and only, and the problem went away

### Expected Behavior

I clicked a valid link in my e-mail server (netzero.com), which, indirectly, should have gone to Instagram.com to allow me to "follow" the given page, of which I had given permission to send me the e-mail in the first place. Here is the very long link (no worries about identity theft, since my logon ID and password are required, but, not included, within):
https://ct.klclick.com/f/a/8l5FcYCvYuwkzuC_zrqmRg~~/AASl5QA~/RgRjsLEyP0TkaHR0cHM6Ly93d3cuaW5zdGFncmFtLmNvbS9icmF2ZXJvYm90Lz91dG1fc291cmNlPUFsbCUyMFN1YnNjcmliZXJzJnV0bV9tZWRpdW09ZW1haWwmdXRtX2NhbXBhaWduPVNBJTIwJTdDJTIwVmVnYW51YXJ5JTIwJTIzMiUyMCVFMiU4NiU5MiUyMCU3QyUyMDEyLjMwLjIwMjElMjAlMjhSdHkzR1glMjkmX2t4PS1zREFraWQxejNoby1feVI2dkxXNDdoZjVyYU9iNWFKc3huRG84bjBBX2MlM0QuWDRkRUNNVwNzcGNCCmG7MizOYYlrcMJSFGVzcGljaV8yQG5ldHplcm8ubmV0WAQAAdkz

### Actual Behavior

The screen flashed, briefly, but, otherwise, nothing happened, so I knew that Adguard was preventing it, for some reason. I thought hitting the ctrl or similar button(s simultaneously) would bypass the block, but, it didn't, and there is no way to say, "Allow this website to follow links" (n.b., enabling that functionality in the Adguard options page did not work, either). I was able to copy the link, paste it into a new browser tab, and it worked, just fine, which proves that the problem wasn't with my anti-virus program, AVG.

### Screenshots

Wouldn't be of any help.

<details><summary>Screenshot:</summary>

</details>

**Comment by Sergey-Lyapin** [comment]
Hello @skaizun
Could you open the page where this link is located, open the AdGuard's filtering log, reproduce the problem, make a screenshot of the filtering log and send it here?

**Comment by Birbber** [comment]
@skaizun Any news?

**Comment by skaizun** [comment]
@Sergey-Lyapin:
As you can, hopefully, see from the image, the resulting Ad Guard log is quite large (note the scroll bar on the right in the image). However, the vast majority of the remainder is for images _(n.b., as you're, probably, aware, AG allows users to select groups of items, but not to de-select/ignore items)._ With any luck, this screen capture will show whatever it is you need _(it would have been nice if AG added a "copy-to-text" function!)._ For the record, I deleted the original link-fail-by-AG e-mail, which had a lot of images and irrelevant-to-this-issue links and junk, so, I created a new e-mail with just the offending link, which "failed" the same way as the original e-mail.

![adguard fail](https://user-images.githubusercontent.com/96894551/149775296-5d09de9e-8c07-4060-aea2-ee4b40a84866.jpg)

**Comment by skaizun** [comment]
If the above image is illegible, in Windows, right-click it, then click "Open in new tab (or window)", and then either left-click to zoom in a bit or use your browser or mouse "zoom" functionality to see it better.

**Comment by Sergey-Lyapin** [comment]
Hello @skaizun, thank you for the screenshot.
Is this the only blocked request? You can see this by selecting "blocked" checkbox. 
Please take a screenshot of the blocked requests.

Try to add this rule to user rules and check link functionality.
`@@||webmailb.uolstatic.com/js/new/adServing.js`

**Comment by skaizun** [comment]
Following is the screenshot you requested. Please note that I deleted the original e-mail, and that I am using one that I created and sent to myself whose sole message content is the link in question, which does not open, per my original message. I tried the user rule you listed, above, but, it didn't seem to affect anything, even after I refreshed the e-mail web page (i.e., the link is still blocked), so, I deleted the user rule.


![adguard problem](https://user-images.githubusercontent.com/96894551/149829795-3381082a-aed1-48f1-8609-49f7f09c853d.jpg)

**Comment by Sergey-Lyapin** [comment]
@skaizun
Thank you, please add this rule to user rules:
`@@||klclick.com^`
and check again

**Comment by skaizun** [comment]
The "klclick.com" user rule WORKED!!!
Now I know how to do it for other such, should they occur!
Thanks, very much!

Both Peter Lowe and Easy Privacy filters are de-selected (i.e., both show an orange 'X').
I presume those are default settings, since I never touched them.

**Comment by Sergey-Lyapin** [comment]
@skaizun
I'm glad to read it, you are welcome =) Do you use AG French filter? It seems it causes the issue.

We need to make the rule more specific to add it to our filters,
could you check the rule below instead of `@@||klclick.com^`?

`@@||klclick.com^$domain=netzero.net`

**Comment by skaizun** [comment]
The AdGuard French filter is enabled.
I removed the shorter/working user rule, and disabled the French filter,
and it worked (i.e., the e-mail link was allowed).
But, since the filter was, presumably, defaulted enabled, I'm leaving it on.

I then added the the _domain_ part **failed** (i.e., the e-mail link was blocked),
so, I re-inserted the shorter user rule, which works, fine.

I consider the matter closed!

Again, thanks very much!
